### PR TITLE
Import Global statements

### PR DESCRIPTION
**Pull request by Thirumalai-Shaktivel**

### CODE REVIEW

**Comment by Thirumalai-Shaktivel** [comment]
TODO
- [ ] Enclose all the global statements into a function: `<module_name>@global_statements`
- [ ] Call the above function using SubroutineCall
- [ ] Global lists initialization has to be handled
- [ ] check multiple `import a` statements
- [ ] Pass all the tests

File: src/lpython/semantics/python_ast_to_asr.cpp [hunk 1]
```diff
@@ -2239,13 +2244,17 @@ class CommonVisitor : public AST::BaseVisitor<Struct> {
...
             ASR::expr_t* v_expr = ASRUtils::EXPR(ASR::make_Var_t(al, loc, v_sym));
             cast_helper(v_expr, init_expr, true);
             ASR::asr_t* assign = ASR::make_Assignment_t(al, loc, v_expr,
                                                         init_expr, nullptr);
-            current_body->push_back(al, ASRUtils::STMT(assign));
+            if (current_body) {
+                current_body->push_back(al, ASRUtils::STMT(assign));
+            } else if (ASR::is_a<ASR::List_t>(*type)) {
+                global_init.push_back(al, assign);
+            }
```
Comment (Thirumalai-Shaktivel) [inline src/lpython/semantics/python_ast_to_asr.cpp:2257]: I assumed `current_body` contains `true` value only when `create_add_variable_to_scope` (this function) is visited inside the `Function` or `Program` nodes. In the module or global scope, `current_body` would be `nullptr`.
So, I have added a check for `global_lists`.
Have I handled this correctly? @czgdp1807

**Comment by Thirumalai-Shaktivel** [comment]
This PR is ready for review.

File: src/lpython/semantics/python_ast_to_asr.cpp (hunk 1 above)
Comment (czgdp1807) [reply src/lpython/semantics/python_ast_to_asr.cpp:2257]: Make sense to me.

**Comment by czgdp1807** [comment]
Please resolve the conflicts.

**Comment by Thirumalai-Shaktivel** [comment]
Done!

File: integration_tests/global_syms_03_a.py [hunk 2]
```diff
@@ -4,4 +4,8 @@
 
 def populate_lists() -> list[i32]:
     return [10, -20]
-l_2: list[i32] = populate_lists()
+l_2: list[i32]
+l_2 = populate_lists()
```
Comment (czgdp1807) [inline integration_tests/global_syms_03_a.py:8]: Why this change?

Comment (Thirumalai-Shaktivel) [reply integration_tests/global_syms_03_a.py:8]: I wanted to add an assignment statement for the initializer. Instead of adding a new test, I just used the existing one 😅.

File: src/libasr/asr_scopes.cpp [hunk 3]
```diff
@@ -206,47 +203,6 @@ void SymbolTable::move_symbols_from_global_scope(Allocator &al,
                 es->m_parent_symtab = module_scope;
                 ASR::symbol_t *s = ASRUtils::symbol_get_past_external(a.second);
                 LCOMPILERS_ASSERT(s);
-                if (ASR::is_a<ASR::Variable_t>(*s)) {
```
Comment (Thirumalai-Shaktivel) [inline src/libasr/asr_scopes.cpp:209]: All the initializers of list types are moved into the global_initializer, so I removed all the changes which do the same thing.

File: integration_tests/global_syms_03_a.py (hunk 2 above)
Comment (czgdp1807) [reply integration_tests/global_syms_03_a.py:8]: I would recommend to have both tests, initialiser and assignment statement for the initialiser.

Comment (Thirumalai-Shaktivel) [reply integration_tests/global_syms_03_a.py:8]: Got it! I will update the tests.

**Review by certik — APPROVED** [review]
I think this looks good to me. Thanks @Thirumalai-Shaktivel !
